### DISCUSSION TITLE
Access to HttpContext in ICacheHash

### DISCUSSION THREAD

**Discussion by vnbaaij** [SixLabors/ImageSharp.Web#178] [Q&A] [answered]
I'm migrating my Episerver (now Optimizely) ImageProcessor package (+222k downloads!) to their new .NET 5 version by using ImageSharp.Web. 
I need to create my own hash and would like to have access to the actual HttpContext (the value parameter does not contain the whole Url). Is this something you would consider with a next version?

I'm doing this because I want to store each cached item in the same folder/container as the source file. Abusing the hash to transfer the container name to my IImageCache derived class... Any  suggestion to achieve this with a more standard approach appreciated!

**Comment by deanmarcussen** [comment, marked as answer]
Can you not just inject the `IHttpContextAccessor` into your custom implementation?

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-context?view=aspnetcore-5.0#use-httpcontext-from-custom-components

**Reply by vnbaaij** [reply]
Thanks! By using this I don't actually need to have an own ICacheHash implementation. I can do what I need directely in my IImageCache implementation.

**Comment by JimBobSquarePants** [comment]
>I'm doing this because I want to store each cached item in the same folder/container as the source file. 

Gotta say this seems like a really odd setup. What about multiple sources?

**Reply by vnbaaij** [reply]
Optimizely (Episerver) works with BLOB storage (https://world.episerver.com/documentation/developer-guides/optimizely-platform/-net-core-preview/CMS/blob-storage-and-providers/) for storing media such as images. There are API's to make it easy to go from the image request to the correct storage container.
I'm using the native functionality to read/write blobs as much as possible and, the same as with the .NET Framework version, I'm storing the cached images in the same container as the source blobs. 

For static files there is one reserved container for caching created in the blob storage. Other sources are not supported (by me)
